src/codecoverage/bot/code_coverage_bot/uploader.py

```diff
@@ -91,3 +94,30 @@ def check_codecov_job():
         return utils.retry(check_codecov_job, retries=30)
     except TotalsNoneError:
         return False
+
+
+def gcp_upload(path, data):
+    '''
+    Upload a data payload on Google Cloud Storage
+    '''
+    assert isinstance(path, str)
+    assert isinstance(data, (bytes, str))
+
+    # Load credentials from Taskcluster secret
+    creds = secrets[secrets.GOOGLE_CLOUD_STORAGE]
+    if 'bucket' not in creds:
+        raise KeyError('Missing bucket in GOOGLE_CLOUD_STORAGE')
+    bucket = creds.pop('bucket')
+
+    # Write temporary file for client creation
```

Comment:
There is no way to pass the creds directly without using a file?

Reply:
Yes and no: there is a low level API to build a Credentials, but it did not work when populating the StorageClient

Reply:
How weird... Anyway, not so important.

Reply:
I got  it working without the file dump ! Now that i have a fully working credentials, it's easier to debug :smirk: